src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivityContext.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// ActivityContext representation conforms to the w3c TraceContext specification. It contains two identifiers
+    /// a TraceId and a SpanId - along with a set of common TraceFlags and system-specific TraceState values.
+    /// </summary>
+    public readonly struct ActivityContext
+    {
+        /// <summary>
+        /// Construct a new object of ActivityContext.
+        /// </summary>
+        /// <param name="traceId"> A trace identifier.</param>
+        /// <param name="spanId"> A span identifier </param>
+        /// <param name="traceFlags">Contain details about the trace.</param>
+        /// <param name="traceState"> Carries system-specific configuration data.</param>
+        public ActivityContext(ActivityTraceId traceId, ActivitySpanId spanId, ActivityTraceFlags traceFlags, string? traceState = null)
+        {
+            TraceId = traceId;
+            SpanId = spanId;
+            TraceFlags = traceFlags;
+            TraceState = traceState;
+        }
+
+        /// <summary>
+        /// The trace identifier
+        /// </summary>
+        public ActivityTraceId TraceId { get; }
+
+        /// <summary>
+        /// The span identifier
+        /// </summary>
+        public ActivitySpanId SpanId { get; }
+
+        /// <summary>
+        /// The flags for the details about the trace.
+        /// </summary>
+        public ActivityTraceFlags TraceFlags { get; }
+
+        /// <summary>
+        /// system-specific configuration data.
+        /// </summary>
+        public string? TraceState { get; }
+
+        internal bool IsValid => TraceId != default && SpanId != default;
```

Comment:
do we need it? I don't see the scenario and suggest that users just check `ActivityContext != default`

Reply:
This is internal and not exposed. I don't think ActivityContext != default would be enough as I believe you can create a context with default TraceId and SpanId and some values for flags and state. This will make ActivityContext != default and at the same time not valid context I guess.

Reply:
I addressed this in https://github.com/dotnet/runtime/pull/32660/commits/7260ecf4917ab68aa1415e7b90b60cdcdfe20ad0